File: sample/anpy/test/cases/sieve.py
# Sieve of Eratosthenes (ChocoPy style: pre-sized list, no append).
n:int = 50
flags:[bool] = None
i:int = 0
j:int = 0
count:int = 0

flags = [True]
# grow to n+1 entries by concatenation
while len(flags) <= n:
    flags = flags + [True]

i = 2
while i * i <= n:
    if flags[i]:
        j = i * i
        while j <= n:
            flags[j] = False
            j = j + i
    i = i + 1

i = 2
while i <= n:
    if flags[i]:
        count = count + 1
    i = i + 1
print(count)

File: sample/rubyharness/bench/sieve.rb
def bench
  n = 50_000
  sieve = Array.new(n + 1, true)
  sieve[0] = sieve[1] = false
  i = 2
  while i * i <= n
    if sieve[i]
      j = i * i
      while j <= n
        sieve[j] = false
        j += i
      end
    end
    i += 1
  end
  count = 0; k = 0
  while k <= n
    count += 1 if sieve[k]
    k += 1
  end
  count
end

result = 0
i = 0
while i < 100
  result = bench
  i += 1
end
p result
